### ISSUE TITLE
Doesn't work in 2.2.4

### ISSUE THREAD

**Issue by numipinto** [NOD507/SenseDateRangePicker#3]
This DataRangePicker is fantastic, but it doesn't work in 2.2.4 version of Qlik Sense.
I believe it's due to the Engine API call that may have changed.

**Comment by stefanwalther** [comment]
Hi, there was no change in the EngineAPI in 2.2.4 which could actually influence the behaviour of this great visualization extension.
Can you pls add some more details what's actually happening.

Does it
- just not work, but still show (so you still see the object and the property panel)
- do you get an "Invalid visualization"
- ?

Regards
Stefan

**Comment by numipinto** [comment]
Hello Stefan,

In terms of user interaction the datepicker works well but I've been debugging the datepicker.js in chrome console and when you select a range or a period the Api request simply has no effect on the visualization.
Tomorrow I'll be working on it again so I can send you some screen shots in order for you to better understand the issue.
Thanks for looking into it.

Best regards,
Nuno

Enviado do meu iPhone

No dia 07/06/2016, às 21:15, Stefan Walther <notifications@github.com> escreveu:

**Comment by numipinto** [comment]
Hello Stefan,

Sorry for the delay but I've been overwhelmed with work and didn't have time to continue "this investigation".
While debugging the JS I've come across the request that is asked of the API.
Is it supposed to be as you can see by the following image?

In the meanwhile I'll keep digging in to better understand why this is happening :)

![image](https://cloud.githubusercontent.com/assets/3770860/16048696/f4330b3e-324c-11e6-83be-078be713f667.png)

Best regards,
Nuno

**Comment by clark116** [comment]
Hi Nuno,

Are you saying that when you select a range, the pop-up disappears and then NOTHING ELSE happens? The API call is sent/received, but no selections are made?

Because as of right now, nothing happens if there are no matching dates for the range selected. In our app, we have no records with sign-up dates in the future -- so the popup just closes if you select anything beyond 'today.'

Try selecting a very broad date range, one that **should** make a selection.

**Comment by numipinto** [comment]
Hello Clark,

In response to your questions:
Yes after a range selection the calendar pop-up disappears and nothing happens to the visualization.
The API call is sent/received even though no selections are made.

I've tried a better range selection as you can see by the image bellow, but still nothing happens.
![image](https://cloud.githubusercontent.com/assets/3770860/16148416/615e9fbc-3481-11e6-9ecf-cf2f3ea83b00.png)

I've been investigating the backend API request search (https://help.qlik.com/en-US/sense-developer/2.2/Subsystems/APIs/Content/BackendAPI/search-method.htm) do you know if it's somehow related to Qlik Sense' smart search ?

Thanks for the help!

**Comment by numipinto** [comment]
Clark and Stefan,

I've discovered why this was happening. The dimension I used in the datepicker is a _timestamp_ (it has the following format: **"DD/MM/YYYY HH:mm:ss"**) which invalidates the search with date format ("DD/MM/YYYY").
For test purposes, I've 'hacked' your _DateRangePicker.js_ in order to force the date search to contemplate also the time part, and it worked.
I've also noticed that _lib\daterangepicker.js_ has time attributes that can be used in to address 'timestamp cases' like mine.

Thanks for all your replies.

Best Regards,
Nuno Pinto
